syft/grid/__init__.py

```diff
@@ -3,20 +3,27 @@
 
 DEFAULT_NETWORK_URL = "ws://ec2-13-59-45-128.us-east-2.compute.amazonaws.com"
 
+_registered_peer = None
+
 
 def register(**kwargs):
     """ Add this process as a new peer registering it in the grid network.
-        
+
         Returns:
             peer: Peer Network instance.
     """
-    if not kwargs:
-        args = args = {"max_size": None, "timeout": 444, "url": DEFAULT_NETWORK_URL}
-    else:
-        args = kwargs
+    global _registered_peer
 
-    peer_id = str(uuid.uuid4())
-    peer = Network(peer_id, **args)
-    peer.start()
+    if _registered_peer is None:
+        if not kwargs:
+            args = args = {"max_size": None, "timeout": 444, "url": DEFAULT_NETWORK_URL}
+        else:
+            args = kwargs
+
+        peer_id = str(uuid.uuid4())
+        _registered_peer = Network(peer_id, **args)
+        _registered_peer.start()
+    else:
+        raise UserWarning(f"You're already a registered peer {_registered_peer}")
```

Comment:
I was wondering if there is a need to throw an error in the user's face if he tries to register multiple times. I think just returning the registered peer for the first time would be okay.  What do you think?

Reply:
Yep, I think we can return the registered peer instance created by the first call. But we probably need to print a warning (not an exception or error message) to clarify to the user that they already have a registered peer.